packages/lit-html/src/directives/until.ts

```diff
@@ -51,23 +63,44 @@ class UntilDirective extends AsyncDirective {
 
       // We have a Promise that we haven't seen before, so priorities may have
       // changed. Forget what we rendered before.
-      this._lastRenderedIndex = _infinity;
+      this.__lastRenderedIndex = _infinity;
       previousLength = 0;
 
-      Promise.resolve(value).then((resolvedValue: unknown) => {
-        const index = this._values.indexOf(value);
-        // If state.values doesn't contain the value, we've re-rendered without
-        // the value, so don't render it. Then, only render if the value is
-        // higher-priority than what's already been rendered.
-        if (index > -1 && index < this._lastRenderedIndex) {
-          this._lastRenderedIndex = index;
-          this.setValue(resolvedValue);
+      // Note, the callback avoids closing over `this` so that the directive
+      // can be gc'ed before the promise resolves
```

Comment:
Nit: do we want to copy this comment to the `AsyncReplaceDirective`, too?

Reply:
+1 Done